backend-rust/src/bin/ccdscan-api.rs

```diff
@@ -155,26 +214,39 @@ async fn main() -> anyhow::Result<()> {
             }
             cancel_token.cancel();
         }
+        result = &mut node_collector_task => {
+            error!("Node collector task stopped.");
+            if let Err(err) = result? {
+                error!("Node collector error: {}", err);
+            }
+            cancel_token.cancel();
+        }
     }
     info!("Shutting down");
     // Ensure all tasks have stopped
-    let _ = tokio::join!(monitoring_task, queries_task, pgnotify_listener);
+    let _ = tokio::join!(monitoring_task, queries_task, pgnotify_listener, node_collector_task);
     Ok(())
 }
 
+#[derive(Clone)]
+struct HealthState {
+    pool:                 sqlx::PgPool,
+    node_status_receiver: Receiver<Option<Vec<NodeStatus>>>,
```

Comment:
Consider introducing a type alias `type Receiver = Receiver<Option<Vec<NodeStatus>>` in `graphql_api/node_status.rs`.